deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -0,0 +1,663 @@
+                attributes = self.getAttributes()
+                data = {"path": self._file, "mode": self.event_mode, "type": self.event_type, "timestamp": timestamp, "attributes": attributes}
+            if self.event_type == "modified":
+                timestamp = int(time())
+                self.generateAttributes()
+                attributes = self.getAttributes()
+                self.generateAttributes()
+                old_attributes = self.getAttributes()
+                changed_attributes = self.checkChangedAttributes(attributes, old_attributes)
+                data = {"path": self._file, "mode": self.event_mode, "type": self.event_type, "timestamp": timestamp, "attributes": attributes, "old_attributes": old_attributes, "changed_attributes": changed_attributes}
+
+            if self.event_type == "deleted":
+                timestamp = int(time())
+                self.generateAttributes()
+                attributes = self.getAttributes()
+                data = {"path": self._file, "mode": self.event_mode, "type": self.event_type, "timestamp": timestamp, "attributes": attributes}
+
+            message = json.dumps({"type": "event", "data": data})
+
+        if self.agent_version == "3.11":
+            self.generateAttributes()
+            message = '{0}:{1}:{2}:{3}:{4}:{5}:{6}:{7}:{8}:{9} {10}'.format( \
+                   self._size, self._mode, self._uid, self._gid, self._md5, \
+                   self._sha1, self._uname, self._gname, self._mdate, \
+                   self._inode, self._file)
+
+        return self.formatMessage(message)
+    def getMessage(self, event_mode=None, event_type=None):
+        if event_mode is not None:
+            self.event_mode = event_mode
+        else:
+            self.event_mode = choice(["real-time", "whodata", "scheduled"])
+
+        if event_type is not None:
+            self.event_type = event_type
+        else:
+            self.event_type = choice(["added", "modified", "deleted"])
+
+        return self.generateMessage()
+
+class Sender:
+    def __init__(self, manager_address, manager_port="1514", protocol="tcp"):
+        self.manager_address = manager_address
+        self.manager_port = manager_port
+        self.protocol = protocol
+        if self.protocol == "tcp":
+            self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            self.socket.connect((self.manager_address, int(self.manager_port)))
+        if self.protocol == "udp":
+            self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+
+    def sendEvent(self, event):
+        if self.protocol == "tcp":
+            length = pack('<I', len(event))
+            self.socket.send(length+event)
+            #return(self.socket.recv(2048))
+            #self.socket.close() # Not closing. It will close the socket on Ctrl+C.
+        if self.protocol == "udp":
+            self.socket.sendto(event, (self.manager_address, int(self.manager_port)))
+        return(0)
```

Comment:
In python, when not specified otherwise, all functions return a None. None of the uses of this function checks the result so we can remove this